src/ripple/protocol/impl/tokens.cpp

```diff
@@ -255,28 +490,75 @@ decodeBase58 (std::string const& s,
 */
 template <class InverseArray>
 static
-std::string
-decodeBase58Token (std::string const& s,
-    TokenType type, InverseArray const& inv)
+boost::optional<std::pair<Slice, ExtraB58Encoding>>
+decodeBase58Token(
+    Slice s,
+    TokenType type,
+    MutableSlice result,
+    InverseArray const& inv,
+    DecodeBase58Detail::bitset flags)
 {
-    auto ret = decodeBase58(s, inv);
-
-    // Reject zero length tokens
-    if (ret.size() < 6)
-        return {};
-
-    // The type must match.
-    if (type != static_cast<TokenType>(ret[0]))
+    // additional flags based on token type
+    switch (type)
+    {
+        case TokenType::NodePrivate:
+        case TokenType::AccountSecret:
+        case TokenType::FamilyGenerator:
+        case TokenType::FamilySeed:
+            flags.set(DecodeBase58Detail::maybeSecret);
+            break;
+        default:
+            break;
+    }
+    auto r = decodeBase58(s, result, inv, flags);
+    if (!r)
         return {};
+    Slice decoded;
+    DecodeMetadata metadata;
+    std::tie(decoded, metadata) = *r;
+    ExtraB58Encoding extraB58Encoding = ExtraB58Encoding::None;
+
+    if (type == TokenType::FamilySeed &&
+        metadata.encodingType[0] == std::uint8_t(0xE1) &&
```

Comment:
this is duplicated in isRippleLibEncoded(), can it be shared somewhere? i.e. I don't where these constants came from, but at least don't duplicate them :)

Reply:
Oh, you could just call the function on metaData, it was the same class :) `metaData.isRippleLibEncoded()`

Reply:
Addressed in 91c6c712b0973af30980b849d50e33a4f715d449